core/dbt/parser/generic_test_builders.py

```diff
@@ -182,6 +157,45 @@ def __init__(
             if short_name != full_name and "alias" not in self.config:
                 self.config["alias"] = short_name
 
+    def _process_legacy_args(self):
+        config = {}
+        for key in self.CONFIG_ARGS:
+            value = self.args.pop(key, None)
+            if value and "config" in self.args and key in self.args["config"]:
+                raise SameKeyNestedError()
+            if not value and "config" in self.args:
+                value = self.args["config"].pop(key, None)
+            if value is not None:
+                config[key] = value
+
+        return self._render_values(config)
+
+    def _process_config_args(self, config_dict: Dict[str, Any]) -> Dict[str, Any]:
+        config = {}
+        for key, value in config_dict.items():
+            if value is not None:
```

Comment:
Same as above, this check is happening in `self._render_values`. And since that's all this method is doing, you could drop it then, and update line 133-135 to:
```python
self.config.update(self._render_values(self.args.pop("config", {}))
```